src/generic_ui/scripts/user.ts

```diff
@@ -41,46 +40,61 @@ module UI {
       }
       this.name = profile.name;
       this.imageData = profile.imageData || UI.DEFAULT_USER_IMG;
-      this.isOnline = profile.isOnline;
     }
 
     // Returns two strings, where each matches an array name in model.contacts.
     // Does not use an enum because we need a string value, and typescript
     // enums evaluate to numbers.
```

Comment:
We need strings because we use these category strings to index into model.contacts.getAccessContacts and model.contacts.shareAccessContacts.

i.e. model.contacts.getContacts is map from strings like "onlineTrustedUproxy" and "offlineTrustedUproxy" to arrays.  We can then write code like:
var categories = user.getCategories();
model.contacts.getAccessContacts[categories.getTab].push(user);  // after removing user from their old category, where categories.getTab might evaluate to "onlineTrustedUproxy"

If we have a typescript enum like:
enum Categories { onlineTrustedUproxy = 1000; offlineTrustedUproxy; ... }
and return Categories.onlineTrustedUproxy, this will actually return 1000, which we can't use to index into model.contacts.getAccessContacts.

We've had similar problems with mixing typescript and freedom enums (this is the reason we have to convert the clientState given to us by freedom to our own uproxy clientState object) - in general typescript enums have not been good for us.